etc/config-template.toml

```diff
@@ -64,6 +64,9 @@
 ## If not set, `addr` will be used.
 # advertise-addr = ""
 
+## HTTP listening address.
+# http-addr = "127.0.0.1:0". If it is not specified, the system will use a random port.
```

Comment:
If `:0` is used, how would the user know which random port is chosen?

Reply:
IMO, if the user don't set this configuration, it maybe represent that he don't need this feature. In this case, we don't care about which port is listening. Does it make sense? @kennytm

Reply:
Hmm perhaps don't set up the HTTP server at all in this case?

Reply:
Don't we need a default configuration that will make metrics available for the dashboard? Can't we pick a standard port for our configuration?

Reply:
How does TiDB handle the situation? I think it's better to keep behavior consistent.

Reply:
TiDB allows configuring the port only, not the whole SocketAddress. 

Relevant part of the config sample:

https://github.com/pingcap/tidb/blob/36bcf5db4ab610aff7287081da04e2893760a45a/config/config.toml.example#L109-L120